hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
 	}
 
 	private void checkTransactionNeeded() {
-		if ( !isTransactionInProgress() ) {
+		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
```

Comment:
ok, keep it simple!
I was just wondering, but indeed there's no reason to change it until someone reports it.